File: src/main/kotlin/com/ak/be/engine/db/entity/AkMenuEntity.java
package com.ak.be.engine.db.entity;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.Collection;
import java.util.Objects;

@Entity
@Table(name = "AK_MENU", schema = "PUBLIC", catalog = "DEFAULT")
public class AkMenuEntity {
    private int id;
    private Timestamp createdAt;
    private Timestamp updatedAt;
    private int state;
    private int type;
    private AkDishEntity akDishByDishId;
    private AkRestaurantEntity akRestaurantByRestaurantId;
    private Collection<AkOrderEntity> akOrdersById;

    @Id
    @Column(name = "ID")
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "CREATED_AT")
    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    @Basic
    @Column(name = "UPDATED_AT")
    public Timestamp getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Timestamp updatedAt) {
        this.updatedAt = updatedAt;
    }

    @Basic
    @Column(name = "STATE")
    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;
    }

    @Basic
    @Column(name = "TYPE")
    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AkMenuEntity that = (AkMenuEntity) o;
        return id == that.id &&
                state == that.state &&
                type == that.type &&
                Objects.equals(createdAt, that.createdAt) &&
                Objects.equals(updatedAt, that.updatedAt);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, createdAt, updatedAt, state, type);
    }

    @ManyToOne
    @JoinColumn(name = "DISH_ID", referencedColumnName = "ID", nullable = false)
    public AkDishEntity getAkDishByDishId() {
        return akDishByDishId;
    }

    public void setAkDishByDishId(AkDishEntity akDishByDishId) {
        this.akDishByDishId = akDishByDishId;
    }

    @ManyToOne
    @JoinColumn(name = "RESTAURANT_ID", referencedColumnName = "ID", nullable = false)
    public AkRestaurantEntity getAkRestaurantByRestaurantId() {
        return akRestaurantByRestaurantId;
    }

    public void setAkRestaurantByRestaurantId(AkRestaurantEntity akRestaurantByRestaurantId) {
        this.akRestaurantByRestaurantId = akRestaurantByRestaurantId;
    }

    @OneToMany(mappedBy = "akMenuByMenuId")
    public Collection<AkOrderEntity> getAkOrdersById() {
        return akOrdersById;
    }

    public void setAkOrdersById(Collection<AkOrderEntity> akOrdersById) {
        this.akOrdersById = akOrdersById;
    }
}

File: src/main/kotlin/com/ak/be/engine/db/entity/AkMenuEntity.kt
package com.ak.be.engine.db.entity

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.sql.Timestamp
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "AK_MENU", schema = "PUBLIC", catalog = "DEFAULT")
class AkMenuEntity {
    @get:Id
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    @get:Column(name = "ID")
    var id: Int = 0
    @get:Basic
    @get:CreationTimestamp
    @get:Column(name = "CREATED_AT")
    var createdAt: Timestamp? = null
    @get:Basic
    @get:UpdateTimestamp
    @get:Column(name = "UPDATED_AT")
    var updatedAt: Timestamp? = null
    @get:Basic
    @get:Column(name = "STATE")
    var state: Int = 0
    @get:Basic
    @get:Column(name = "TYPE")
    var type: Int = 0
    @get:ManyToOne
    @get:JoinColumn(name = "DISH_ID", referencedColumnName = "ID", nullable = false)
    var akDishByDishId: AkDishEntity? = null
    @get:ManyToOne
    @get:JoinColumn(name = "RESTAURANT_ID", referencedColumnName = "ID", nullable = false)
    var akRestaurantByRestaurantId: AkRestaurantEntity? = null
    @get:OneToMany(mappedBy = "akMenuByMenuId")
    var akOrdersById: Collection<AkOrderEntity>? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as AkMenuEntity?
        return id == that!!.id &&
                state == that.state &&
                type == that.type &&
                createdAt == that.createdAt &&
                updatedAt == that.updatedAt
    }

    override fun hashCode(): Int {
        return Objects.hash(id, createdAt, updatedAt, state, type)
    }
}
